test/examples/src/index.md

```diff
@@ -228,6 +228,28 @@ julia> f()
 foobuu
 ```
 
+## Hiding source lines in doctests
+
+```jldoctest
+A = 1
+A += 1 # hide
+A
+
+# output
+
+2
+```
```

Comment:
This example made me question for an instant if we should allow `#hide` in doctests in the first place. I do think it is good to be consistent with at-example and it can be useful for hiding some repetitive initialization etc. But the other side of it is that the users should probably be able to run the doctests as they appear in the manual.

As a compromise, should we maybe add a note in the docs suggesting to the user to use `#hide` sparingly, to make sure they're not hiding relevant parts of the input?

Reply:
I agree that this is a bad example and this is not the usecase I have in mind. I am not a fan of disallowing things though, if the user wants to write bad example that is fine with me :smile: But yea, we can add a note about that.

The real usecase I have for this is that I essentially want a doctestsetup that is not hidden. Consider something like:

````
```jldoctest
foo(a, b) = a + b
bar(a, b) = a * b
```

Something something, with the definitions above we can do this cool stuff:

```jldoctest
julia> foo(1, 2)
3

julia> bar(2, 3)
6
```
````

So in the first block there I don't want to compare the output (since it would be very ugly...). This is where I imagined appending
```
nothing # hide

# output

```
to the first block, we can remove the `hide` line, and since we return `nothing` it will compare correctly against the empty output. And then we remove empty output sections...

While I am writing this I feel this might not be the best approach. Perhaps we can allow `jldoctest` blocks without `# output`? Such a block would simply evaluate into the named doctest module, but not capture, or compare, any output. Another option would be to do #652, and then you can use a `@example` block with trailing `nothing # hide` to this end. Not sure...

Reply:
Yea, do you compare the output still?

Reply:
Perhaps https://github.com/JuliaDocs/Documenter.jl/issues/252#issuecomment-389412041